CesiumRasterOverlays/src/RasterOverlayUtilities.cpp

```diff
@@ -875,8 +875,250 @@ T getVertexValue(
+        buffer.cesium.data,
+        bufferView.byteOffset + accessor.byteOffset,
+        accessorByteStride,
+        accessorComponentElements,
+        attribute.second,
+        std::vector<double>(
+            static_cast<size_t>(accessorComponentElements),
+            std::numeric_limits<double>::max()),
+        std::vector<double>(
+            static_cast<size_t>(accessorComponentElements),
+            std::numeric_limits<double>::lowest()),
+    });
+  }
+
+  if (uvAccessorIndex == -1 || positionAccessorIndex == -1 ||
+      positionAttributeCount == -1) {
+    // We don't know how to divide this primitive, so just remove it.
+    return false;
+  }
+
+  for (const std::string& attribute : toRemove) {
+    primitive.attributes.erase(attribute);
+  }
+
+  const bool keepAboveU = !isWestChild(childID);
+  const bool keepAboveV =
+      hasInvertedVCoordinate ? isSouthChild(childID) : !isSouthChild(childID);
+
+  const AccessorView<glm::vec2> uvView(parentModel, uvAccessorIndex);
+  const AccessorView<glm::vec3> posView(parentModel, positionAccessorIndex);
+  if (uvView.status() != AccessorViewStatus::Valid ||
+      posView.status() != AccessorViewStatus::Valid) {
+    return false;
+  }
+
+  std::vector<uint32_t> clipVertexToIndices;
+  std::vector<CesiumGeometry::TriangleClipVertex> clippedA;
+  std::vector<CesiumGeometry::TriangleClipVertex> clippedB;
+
+  std::vector<uint32_t> vertexMap(
+      size_t(uvView.size()),
+      std::numeric_limits<uint32_t>::max());
+
+  std::vector<float> newVertexFloats;
+
+  for (int64_t i = 0; i < posView.size(); i++) {
+    const glm::vec3 pos = posView[i];
+    const glm::vec2 uv = uvView[i];
+
+    const bool isU = keepAboveU ? uv.x > 0.5 : uv.x < 0.5;
+    const bool isV = keepAboveV ? uv.y > 0.5 : uv.y < 0.5;
+
+    if (isU && isV) {
+      newVertexFloats.emplace_back(pos.x);
+      newVertexFloats.emplace_back(pos.y);
+      newVertexFloats.emplace_back(pos.z);
+    }
+  }
+
+  if (newVertexFloats.empty()) {
+    return false;
+  }
+
+  // Update the accessor vertex counts and min/max values
+  const int64_t numberOfVertices =
+      int64_t(newVertexFloats.size()) / vertexSizeFloats;
+  for (const FloatVertexAttribute& attribute : attributes) {
+    Accessor& accessor =
+        model.accessors[static_cast<size_t>(attribute.accessorIndex)];
+    accessor.count = numberOfVertices;
+    accessor.min = attribute.minimums;
+    accessor.max = attribute.maximums;
+  }
+
+  // Populate the buffers
+  Buffer& vertexBuffer = model.buffers[vertexBufferIndex];
+  vertexBuffer.cesium.data.resize(newVertexFloats.size() * sizeof(float));
+  float* pAsFloats = reinterpret_cast<float*>(vertexBuffer.cesium.data.data());
+  std::copy(newVertexFloats.begin(), newVertexFloats.end(), pAsFloats);
+  vertexBuffer.byteLength = vertexBufferView.byteLength =
+      int64_t(vertexBuffer.cesium.data.size());
+  vertexBufferView.byteStride = vertexSizeFloats * int64_t(sizeof(float));
+
+  bool onlyWater = false;
```

Comment:
All this water mask stuff looks duplicated with the triangle version of the function. Can it be factored out? We don't have, and may never have, a POINTS dataset with a water mask, but it's reasonable enough to support it as long as it's easy.